### DISCUSSION TITLE
Should we avoid exposing scopes to `/authorize`?

### DISCUSSION THREAD

**Discussion by XaviTorello** [fastapi-users/fastapi-users#664] [Ideas]
The `/authorize` endpoint needs to receive the scopes to be able to compute the `authorization_url`.

Maybe it's a little bit insecure if we expose this setting directly through the endpoint.

Following [this approach](https://github.com/frankie567/httpx-oauth/discussions/253), the scopes can be configured using env vars, delegating the settings to FastAPI and the `httpx-oauth.BaseOAuth2` subclasses.

Our proposal is to modify the `/authorize` endpoint to avoid receiving the scopes, delegating directly this setting to the OAuth2 backend.

Ideas?

**Comment by frankie567** [comment]
The fact that `/authorize` accepts a list of scope is here on purpose to be able to implement incremental authorization: https://developers.google.com/identity/protocols/oauth2/web-server#incrementalAuth

This way, we can request new access tokens with new scopes dynamically, following what the user is doing on the application ; and avoid to ask for a large list of scopes upfront.

In terms of security, I don't think there is too much risk: either way, the list of scopes is exposed on the provider authorization URL. Besides, even if the user tries to tinker with them, all they will do will be to generate access tokens with larger/smaller scopes.

However, I'm open to accept your [PR on HTTPX-OAuth](https://github.com/frankie567/httpx-oauth/pull/255) to have the default scopes at initialization. Once this is released, no change should be required on FastAPI Users.
